samples/drivers/lora_lbm/ping_pong/src/main_ping_pong.c

```diff
@@ -0,0 +1,250 @@
+#include <zephyr/devicetree.h>
+
+/* #include "apps_common.h" */
+/* #include "sx126x_radio.h" */
+#include "sx126x.h"
+#include "sx126x_hal.h"
+#include "sx126x_regs.h"
+/* #include "sx126x_board.h" */
+#include "main_ping_pong.h"
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(main);
+
+/*
+ * -----------------------------------------------------------------------------
+ * --- PRIVATE MACROS-----------------------------------------------------------
+ */
+
+/**
+ * @brief Size of ping-pong message prefix
+ *
+ * Expressed in bytes
+ */
+#define PING_PONG_PREFIX_SIZE 6
+
+/**
+ * @brief Duration of the wait before packet transmission to assure
+ / reception status is ready on the other side
+ *
+ * Expressed in milliseconds
+ */
+#define DELAY_BEFORE_TX_MS 20
+
+/**
+ * @brief Duration of the wait between each ping-pong activity,
+ * can be used to adjust ping-pong speed
+ *
+ * Expressed in milliseconds
+ */
+#define DELAY_PING_PONG_PACE_MS 200
+
+/**
+ * @brief LR11xx interrupt mask used by the application
+ */
+#define IRQ_MASK                                                                                   \
+	(SX126X_IRQ_TX_DONE | SX126X_IRQ_RX_DONE | SX126X_IRQ_TIMEOUT | SX126X_IRQ_HEADER_ERROR |  \
+	 SX126X_IRQ_CRC_ERROR)
+
+/*
+ * -----------------------------------------------------------------------------
+ * --- PRIVATE CONSTANTS -------------------------------------------------------
+ */
+
+/*
+ * -----------------------------------------------------------------------------
+ * --- PRIVATE TYPES -----------------------------------------------------------
+ */
+
+/*
+ * -----------------------------------------------------------------------------
+ * --- PRIVATE VARIABLES -------------------------------------------------------
+ */
+
+#if DT_HAS_CHOSEN(lora)
+#define LORA_TRANSCEIVER DT_CHOSEN(lora)
+#else
+#define LORA_TRANSCEIVER DT_NODELABEL(lora0)
+#endif
+
+const struct device *context = DEVICE_DT_GET(LORA_TRANSCEIVER);
+const struct device *spi_bus = DEVICE_DT_GET(DT_BUS(LORA_TRANSCEIVER));
+
+static uint8_t buffer_tx[PAYLOAD_LENGTH];
+static bool is_master = true;
+
+static const uint8_t ping_msg[PING_PONG_PREFIX_SIZE] = "M-PING";
+static const uint8_t pong_msg[PING_PONG_PREFIX_SIZE] = "S-PONG";
+
+static const uint8_t prefix_size = MIN(PING_PONG_PREFIX_SIZE, PAYLOAD_LENGTH);
+
+/*
+ * -----------------------------------------------------------------------------
+ * --- PRIVATE FUNCTIONS DECLARATION -------------------------------------------
+ */
+
+/**
+ * @brief Handle reception failure for ping-pong example
+ */
+static void ping_pong_reception_failure_handling(void);
+
+/*
+ * -----------------------------------------------------------------------------
+ * --- PUBLIC FUNCTIONS DEFINITION ---------------------------------------------
+ */
+
+/**
+ * @brief Main application entry point.
+ */
+int main(void)
+{
+	LOG_INF("===== SX126x TX CW example =====");
+
+	apps_common_lr11xx_system_init(context);
```

Comment:
You use `apps_common_` functions here, but there is no `apps_common` module.

Reply:
Ah yes, indeed, that's part of the samples that were copy-pasted from SWL2001, but should be reworked for a cleaner integration in Zephyr.